### ISSUE TITLE
redis-cluster-2 CrashLoopBackOff after deployment

### ISSUE THREAD

**Issue by david-mun** [amaizfinance/redis-operator#14]
If I deploy redis with 3 replica, I am frequently getting CrashLoopBackOff error for redis-cluster-2 but all other pods are running find including operator. When I described the failed pod, I can see the blow logs:

Last State: Terminated
Reason: ContainerCannotRun
Message: OCI runtime create failed: container_linux.go:348: starting container process caused "process_linux.go:402: container init caused \"rootfs_linux.go:58: mounting \\\"/var/lib/kubelet/pods/3ba5043e-ed13-4a47-8554-f29bb14dec94/volume-subpaths/redis-cluster-config/redis/0\\\" to rootfs \\\"/u01/data/docker/overlay2/a75dd75f91d5e32c330de89c49c13e9302447a2b3fe4550c9c48e0c0b8f63dfa/merged\\\" at \\\"/u01/data/docker/overlay2/a75dd75f91d5e32c330de89c49c13e9302447a2b3fe4550c9c48e0c0b8f63dfa/merged/config/redis.conf\\\" caused \\\"no such file or directory\\\"\"": unknown

But if I restart the pod by deleting it, it's running fine.  Any idea why it's failing for the initial deployment? might be some timing issue for creating configMap ?

**Comment by shamil** [comment]
I'm getting similar error, and only recreating the pod helps resolving it.

**Comment by logdnalf** [comment]
I think that perhaps you are hitting the following bug: https://github.com/kubernetes/kubernetes/issues/68211

**Comment by logdnalf** [comment]
Which version of Docker and Kubernetes are you using?

**Comment by nrvnrvn** [comment]
@david-mun @shamil @logdnalf hey guys,

Could you please check if you are still hit with this issue while using the [latest release of redis-operator](https://github.com/amaizfinance/redis-operator/releases/tag/v0.2.0)?

**Comment by shamil** [comment]
@nrvnrvn it's not always reproducible (at least for me), it work ok most of the time, so no complains...

**Comment by shamil** [comment]
@nrvnrvn experienced several times in pas 2 weeks. Getting same issue. Let me know if you want me to provide any information.
Nothing special  in the operator log, and same error message like in @david-mun posted.

```
Error: failed to start container "redis": Error response from daemon: OCI runtime create failed: container_linux.go:349: starting container process caused "process_linux.go:449: container init caused \"rootfs_linux.go:58: mounting \\\"/var/lib/kubelet/pods/ca00f7a8-fc92-4f9e-8a01-8c0a8d1bfc21/volume-subpaths/redis-scheduler-db-config/redis/0\\\" to rootfs \\\"/var/lib/docker/overlay2/6c19374e02bbfad00f6f8ed03fa8e9315af2d9573d9f895ec8c623ea24963adc/merged\\\" at \\\"/var/lib/docker/overlay2/6c19374e02bbfad00f6f8ed03fa8e9315af2d9573d9f895ec8c623ea24963adc/merged/config/redis.conf\\\" caused \\\"no such file or directory\\\"\"": unknown
```

**Comment by step76** [comment]
I've experienced the same error on my cluster using the latest operator version (0.2.1).

**Comment by step76** [comment]
My cluster was upgraded to 1.19 and the error seems solved.

**Comment by nrvnrvn** [comment]
I am thinking about closing this issue because the upstream issue has been resolved as of Kubernetes 1.19.
On the other side the workaround still looks valid and hashing the configmap's data and adding it as an annotation to pods makes perfect sense. See https://github.com/kubernetes/kubernetes/issues/68211#issuecomment-667138868
